src/test/java/com/volasoftware/tinder/UserServiceTest.java

```diff
@@ -414,4 +418,102 @@ public void testFindFriendByIdWithNotFriends() {
+
+        PageRequest pageRequest = PageRequest.of(0, 10, Sort.by(Sort.Order.desc("rating")));
+
+        List<FriendRatingDetails> mockFriends = List.of(mockFriend);
+
+        Authentication authentication = mock(Authentication.class);
+        when(authentication.getName()).thenReturn("someUsername");
+
+        SecurityContext securityContext = mock(SecurityContext.class);
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+        SecurityContextHolder.setContext(securityContext);
+
+        when(userRepository.findOneByEmail(anyString())).thenReturn(Optional.of(loggedUser));
+        when(userRepository.findFriendsSortedByRating(eq(loggedUser.getId()), eq(pageRequest)))
+                .thenReturn(new PageImpl<>(mockFriends, pageRequest, mockFriends.size()));
+
+        Page<FriendRatingDetails> result = userServiceImpl.findFriendsSortedByRating(
+                pageRequest.getPageNumber(), pageRequest.getPageSize());
+
+        assertEquals(1, result.getTotalElements());
+        assertEquals(mockFriend.getFirstName(), result.getContent().get(0).getFirstName());
+        assertEquals(mockFriend.getLastName(), result.getContent().get(0).getLastName());
+        verify(userRepository, times(1)).findFriendsSortedByRating(loggedUser.getId(), pageRequest);
+        verifyNoMoreInteractions(userRepository);
+    }
+
+    @Test
+    public void testFindFriendsSortedByRatingWihNoFriends() {
+        User loggedUser = new User();
+        loggedUser.setId(1L);
+
+        PageRequest pageRequest = PageRequest.of(0, 10, Sort.by("rating").descending());
+
+        Authentication authentication = mock(Authentication.class);
+        when(authentication.getName()).thenReturn("someUsername");
+
+        SecurityContext securityContext = mock(SecurityContext.class);
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+        SecurityContextHolder.setContext(securityContext);
+
+        when(userRepository.findOneByEmail(anyString())).thenReturn(Optional.of(loggedUser));
+
+        when(userRepository.findFriendsSortedByRating(eq(loggedUser.getId()), eq(pageRequest)))
+                .thenReturn(new PageImpl<>(Collections.emptyList()));
+
+        Page<FriendRatingDetails> result = userServiceImpl.findFriendsSortedByRating(
+                pageRequest.getPageNumber(), pageRequest.getPageSize());
+
+        assertEquals(0, result.getTotalElements());
+        assertTrue(result.getContent().isEmpty());
+        verify(userRepository, times(1)).findFriendsSortedByRating(loggedUser.getId(), pageRequest);
+        verifyNoMoreInteractions(userRepository);
+    }
+
+    @Test
+    public void testFindFriendsSortedByRatingWithMultipleFriends() {
+        User loggedUser = new User();
+        loggedUser.setId(1L);
+
+        FriendRatingDetailsImpl mockFriend1 = new FriendRatingDetailsImpl("John", "Doe", 25, 10);
```

Comment:
Just have seen this, can you test this method with different user data, not just for each to be John Doe 25 10, what will happed if we change one of the mocked friends?